src/dhcp/nm-dhcp-dhcpcanon.c

```diff
@@ -0,0 +1,308 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: t; c-basic-offset: 4 -*- */
+/* nm-dhcp-dhcpcanon.c - dhcpcanon specific hooks for NetworkManager
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Copyright (C) 2017 juga <juga at riseup dot net>
+ */
+
+#include <config.h>
+#define __CONFIG_H__
+
+#define _XOPEN_SOURCE
+#include <time.h>
+#undef _XOPEN_SOURCE
+
+#include "nm-default.h"
+
+#if WITH_DHCPCANON
+
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <ctype.h>
+
+#include "nm-utils.h"
+#include "nm-dhcp-manager.h"
+#include "NetworkManagerUtils.h"
+#include "nm-dhcp-listener.h"
+#include "nm-dhcp-client-logging.h"
+
+/*****************************************************************************/
+
+/* TODO: uncomment to merge with current master
+ * static const char *
+ * _addr_family_to_path_part (int addr_family)
```

Comment:
on master, there is now `nm_utils_addr_family_to_char()`. Use that. It gives '4' instead of "" for IPv4, but that seems desired.

Reply:
Thanks, i think then this can also be changed: https://github.com/NetworkManager/NetworkManager/blob/master/src/dhcp/nm-dhcp-dhclient.c#L127

Reply:
nm-dhcp-dhclient.c uses `_addr_family_to_path_part()` which returns "" for AF_INET. It is different from `nm_utils_addr_family_to_char()`, which would return '4'. It needs to stay this way, to preserve previous behavior. But for the new client, let's not do that.

Reply:
Thanks for the clarification.